pkg/core/consensus/initiator/initiator.go

```diff
@@ -54,35 +48,25 @@ func startProvisioner(eventBroker *eventbus.EventBus, rpcBus *rpcbus.RPCBus, w *
 	}
 }
 
-func startBlockGenerator(eventBroker eventbus.Broker, rpcBus *rpcbus.RPCBus, w *wallet.Wallet) {
+// storeBidValues finds the most recent bid belonging to the given
+// wallet, and stores the relevant values needed by the consensus.
+// This allows the components for block generation to properly function.
+func storeBidValues(eventBroker eventbus.Broker, rpcBus *rpcbus.RPCBus, w *wallet.Wallet) {
 	k, err := w.ReconstructK()
 	if err != nil {
 		panic(err)
 	}
 
 	m := zkproof.CalculateM(k)
 	_, db := heavy.CreateDBConnection()
-	for i := 0; ; i++ {
-		// Get block hash for the given height
-		var hash []byte
-		err := db.View(func(t database.Transaction) error {
-			var err error
-			hash, err = t.FetchBlockHashByHeight(uint64(i))
-			return err
-		})
-
-		// We hit the end of the chain, so we should just exit here
+	for i := uint64(0); ; i++ {
+		hash, err := getBlockHashForHeight(db, i)
 		if err != nil {
+			// We hit the end of the chain, so just exit here
```

Comment:
We should strive to differentiate between errors associated with expected behaviors and unexpected errors. Please introduce an error type we can check (and handle differently)